source/add-existing.txt

```diff
@@ -0,0 +1,232 @@
+
+The following code is a sample ``config/application.rb`` file that
+demonstrates how to specify individual frameworks instead of using ``rails/all``:
+
+.. code-block:: ruby
+   :caption: config/application.rb
+
+   # Remove or comment out rails/all
+   #require "rails/all"
+ 
+   # Add the following instead of rails/all:
+   require "rails"
+ 
+   # Comment out unneeded frameworks
+   # require "active_record/railtie" rescue LoadError
+   # require "active_storage/engine" rescue LoadError
+   require "action_controller/railtie" rescue LoadError
+   require "action_view/railtie" rescue LoadError
+   require "action_mailer/railtie" rescue LoadError
+   require "active_job/railtie" rescue LoadError
+   require "action_cable/engine" rescue LoadError
+   # require "action_mailbox/engine" rescue LoadError
+   # require "action_text/engine" rescue LoadError
+   require "rails/test_unit/railtie" rescue LoadError
+
+.. note::
+
+   Because they rely on ActiveRecord, the `ActionText
+   <https://guides.rubyonrails.org/action_text_overview.html>`__,
+   `ActiveStorage <https://edgeguides.rubyonrails.org/active_storage_overview.html>`__ and
+   `ActionMailbox
+   <https://guides.rubyonrails.org/action_mailbox_basics.html>`__
+   adapters cannot be used alongside {+odm+}.
+
+Disable ActiveRecord
+~~~~~~~~~~~~~~~~~~~~
+
+Review your application's configuration files, such as
+``config/application.rb``, then remove or comment out any references to
+``config.active_record`` and ``config.active_storage``.
+
+Adjust Models
+~~~~~~~~~~~~~
+
+If your application already uses models, you must adjust them to
+migrate from using ActiveRecord to {+odm+}.
+
+ActiveRecord models derive from ``ApplicationRecord`` and do not have
+column definitions. {+odm+} models generally have no superclass but must
+include the ``Mongoid::Document`` attribute. {+odm+} models usually
+define fields explicitly. You can also use :ref:`dynamic fields
+<mongoid-dynamic-fields>` instead of explicit field definitions.
+
+For example, a basic ActiveRecord ``Post`` model might resemble the
+following:
+
+.. code-block:: ruby
+   :caption: app/models/post.rb
+
+   class Post < ApplicationRecord
+     has_many :comments, dependent: :destroy
+   end
+
+A similar {+odm+} ``Post`` model might resemble the following:
+
+.. code-block:: ruby
+   :caption: app/models/post.rb
+
+   class Post
+     include Mongoid::Document
+ 
+     field :title, type: String
+     field :body, type: String
+ 
+     has_many :comments, dependent: :destroy
+   end
+
+Or, you can define the ``Post`` model by using dynamic fields, as shown
+in the following code:
+
+.. code-block:: ruby
+   :caption: app/models/post.rb
+
+   class Post
+     include Mongoid::Document
+     include Mongoid::Attributes::Dynamic
+ 
+     has_many :comments, dependent: :destroy
+   end
+
+{+odm+} does not use ActiveRecord migrations, because MongoDB does not
+require a defined schema before you can store data.
```

Comment:
S: maybe my own ignorance, but this feels tacked on to the previous section instead of a logical continuation, so maybe a note admonition would be useful. maybe even move to the next section (or delete altogether), since it makes this same point.

```suggestion
.. note:: ActiveRecord Migrations

   {+odm+} does not use ActiveRecord migrations, because MongoDB does not
   require a defined schema before you can store data.
```

Reply:
Deleted